### PR TITLE
emails: Convert emails to lowercase while storing/searching for users

### PR DESCRIPTION
**Pull request by SibiAkkash**
@punchagan part 2 of #165

### CODE REVIEW

File: server/management/commands/import_uc_registrations.py [hunk 1]
```diff
@@ -80,6 +80,10 @@ def handle(self, *args: Any, **options: Any) -> None:
                 )
             )
             persons_data = [registration["Person"] for registration in registrations]
+
+            for i in range(len(persons_data)):
+                persons_data[i] = persons_data[i]["email_canonical"].lower()
```
Comment (Joe2k) [inline server/management/commands/import_uc_registrations.py:85]: shoudnt this be `persons_data[i]["email_canonical"] = persons_data[i]["email_canonical"].lower()` ?

Comment (SibiAkkash) [reply server/management/commands/import_uc_registrations.py:85]: oh mb joe, i'll fix this

Comment (punchagan) [reply server/management/commands/import_uc_registrations.py:85]: Also, why use range, len instead of just iterating over the elements?

Comment (SibiAkkash) [reply server/management/commands/import_uc_registrations.py]: @punchagan rookie mistake from me. I forgot that by default objects are always passed by reference (mutable) in python

**Comment by punchagan** [comment]
The validator [here](https://github.com/india-ultimate/hub/commit/759092afbe12dce894dc42b5372cae07e76a539e) is nice, but it doesn't really get called unless we call `user.full_clean()` before we call a `save()`. I feel it doesn't do much for us, even when it gives the appearance of preventing DB level issues. I'm going to merge this PR, without that change.

**Comment by SibiAkkash** [comment]
Yes, that was the problem with adding that validator, it would have to be explicitly called. I wonder if there's any method to auto-validate before inserting to the table

**Comment by SibiAkkash** [comment]
> > Yes, that was the problem with adding that validator, it would have to be explicitly called. I wonder if there's any method to auto-validate before inserting to the table
> 
> We could override the save method to convert email to lowercase, but I'm not sure we need to be so paranoid about this. :)

lol yes i did see some blogs about overriding the save() method, but that seemed a bit too much :P

**Pull request merged**